### PR TITLE
mypy optimize_indexing.py

### PR DESCRIPTION
**Pull request by msaroufim**
Fixes #ISSUE_NUMBER

Not sure what to do about this guy yet `def try_to_reduce_precision(node: Any, bounds: Dict[Any, Any], indirect_vars: List[Any], indices: Dict[Any, Any], replacement_vals: Dict[Any, Any]) -> None:`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/optimize_indexing.py [hunk 1]
```diff
@@ -30,17 +31,23 @@ def val_expressable_in_32_bits(val):
...
-def range_expressable_in_32_bits(range):
+def range_expressable_in_32_bits(range: ValueRanges) -> bool:
     return val_expressable_in_32_bits(range.lower) and val_expressable_in_32_bits(
         range.upper
     )
 
 
-def try_to_reduce_precision(node, bounds, indirect_vars, indices, replacement_vals):
+def try_to_reduce_precision(
+    node: Any,
```
Comment (eellison) [inline torch/_inductor/optimize_indexing.py:41]: node here is `fx.Node`

File: torch/_inductor/optimize_indexing.py [hunk 2]
```diff
@@ -30,17 +31,23 @@ def val_expressable_in_32_bits(val):
...
+    bounds: Dict[Any, Any],
```
Comment (eellison) [inline torch/_inductor/optimize_indexing.py:42]: bounds is `Dict[torch.fx.Node, ValueRanges]`

File: torch/_inductor/optimize_indexing.py [hunk 3]
```diff
@@ -30,17 +31,23 @@ def val_expressable_in_32_bits(val):
...
+    indirect_vars: List[Any],
+    indices: Dict[Any, Any],
```
Comment (eellison) [inline torch/_inductor/optimize_indexing.py:44]: [str, sympy.Expr]

File: torch/_inductor/optimize_indexing.py [hunk 4]
```diff
@@ -30,17 +31,23 @@ def val_expressable_in_32_bits(val):
...
     return val_expressable_in_32_bits(range.lower) and val_expressable_in_32_bits(
         range.upper
     )
 
 
-def try_to_reduce_precision(node, bounds, indirect_vars, indices, replacement_vals):
+def try_to_reduce_precision(
+    node: Any,
+    bounds: Dict[Any, Any],
+    indirect_vars: List[Any],
```
Comment (eellison) [inline torch/_inductor/optimize_indexing.py:43]: List[sympy.Symbol]

File: torch/_inductor/optimize_indexing.py [hunk 5]
```diff
@@ -30,17 +31,23 @@ def val_expressable_in_32_bits(val):
...
+    indices: Dict[Any, Any],
+    replacement_vals: Dict[Any, Any],
```
Comment (eellison) [inline torch/_inductor/optimize_indexing.py:45]: `Dict[Union[str, sympy.Symbol], ValueRanges]`

File: torch/_inductor/optimize_indexing.py [hunk 6]
```diff
@@ -30,17 +31,23 @@ def val_expressable_in_32_bits(val):
...
+) -> None:
     # if a downstream use of a node explicitly converts to int32, or float16/float32/float64,
     # then it's precision is set for that chain of uses, and we don't need to consider those
     # dominated values
-    def skip_filter(node):
+    def skip_filter(node: Any) -> bool:
```
Comment (eellison) [inline torch/_inductor/optimize_indexing.py:50]: fx.Node

**Review by eellison — COMMENTED** [review]
looks good but we should try to get actual annotations where it's doable, this one was pretty easy to figure out with a pdb. 

you might also try using monkey type as a tool here

**Comment by eellison** [comment]
mind updating ?

**Comment by msaroufim** [comment]
Ok I had to get clever to get this to pass mypy, hopefully CI passes. This is probably the important part to review https://github.com/pytorch/pytorch/pull/105621/files#diff-e869ca39e024fc20e59f09f627ea4e3ce43aa664ebe9bc40d6b8ca2f201e5ff6R72

**Comment by msaroufim** [comment]
Yeah k nevermind

**Pull request closed without merging**